source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -678,8 +693,14 @@ void JsonTranscoderFilter::buildResponseFromHttpBodyOutput(
   google::api::HttpBody http_body;
   for (auto& frame : frames) {
     if (frame.length_ > 0) {
+      http_body.Clear();
       Buffer::ZeroCopyInputStreamImpl stream(std::move(frame.data_));
-      http_body.ParseFromZeroCopyStream(&stream);
+      if (!HttpBodyUtils::parseMessageByFieldPath(&stream, method_->response_body_field_path,
+                                                  &http_body)) {
+        // TODO(elessar): Return error to client.
+        encoder_callbacks_->resetStream();
```

Comment:
Is there a way to actually return 500 http status code to client right here if first message is invalid and stop fetching bad body from the upstream?

Reply:
you can call encoder_callbackds_->sendLocalReply() to return 500.

Reply:
`sendLocalReply` is only available as part of `decoder_callbacks_`, at encoding phase it's too late to call it.

Reply:
Yes it is usually to late at encodeData, unless you returned StopIteration in encodeHeaders, then you can modify headers and Continue.

Reply:
Is there a way to indicate envoy that there will be no more data (`end_stream=true`)?

Reply:
Yes, end_stream = true means not more data.

Reply:
seems no, we have `StopIterationAndEndStream` in `FilterHeaderStatus` but not in data. cc @mattklein123?

Reply:
Yeah I think this is a new use case we haven't hit before. You might be able to hack something up with `injectEncodedDataToFilterChain` (0 bytes + end_stream), but otherwise I think this would require HCM changes. cc @alyssawilk @snowp  to see if I am forgetting about something.

Reply:
> I think the same way that we have a sendLocalReply for decoder filter callbacks we could have a sendReplyOrResetStream() for encoder filter callbacks which factors out the HCM responseDataTooLarge logic and either ships headers + body directly to the codec, or resets the stream.

Yeah +1. I would probably just submit this PR the way you have it now with a TODO and then circle back with an HCM change if you so desire.